common/core/desktop/tests/unit/kmnkbd/state_api.cpp

```diff
@@ -165,20 +165,20 @@ int main(int, char * [])
   if (attrs->max_context < 16) return __LINE__;
 
   try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_S,
-                                  KM_KBP_MODIFIER_SHIFT));
+                                  KM_KBP_MODIFIER_SHIFT, 1));
   assert(action_items(test_state, {{KM_KBP_IT_CHAR, {0,}, {km_kbp_usv('S')}}}));
   try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_I,
-                                  KM_KBP_MODIFIER_SHIFT));
+                                  KM_KBP_MODIFIER_SHIFT, 1));
   assert(action_items(test_state, {{KM_KBP_IT_CHAR, {0,}, {km_kbp_usv('I')}}}));
-  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_L, 0));
+  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_L, 0, 1));
   assert(action_items(test_state, {{KM_KBP_IT_CHAR, {0,}, {km_kbp_usv('l')}}}));
 
-  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_BKSP, 0));
+  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_BKSP, 0, 1));
   assert(action_items(test_state, {{KM_KBP_IT_BACK, {0,}, {0}}}));
   try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_L,
-                                  KM_KBP_MODIFIER_SHIFT));
+                                  KM_KBP_MODIFIER_SHIFT, 1));
   assert(action_items(test_state, {{KM_KBP_IT_CHAR, {0,}, {km_kbp_usv('L')}}}));
-  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_F2,0));
+  try_status(km_kbp_process_event(test_state, KM_KBP_VKEY_F2, 0, 1));
```

Comment:
We need to be adding keyup events here too.

Reply:
done